python/semantic_kernel/kernel.py

```diff
@@ -421,3 +430,59 @@ async def _inner_auto_function_invoke_handler(self, context: AutoFunctionInvocat
             else:
                 context.function_result = FunctionResult(function=context.function.metadata, value=value)
             return
+
+    async def add_embedding_to_object(
+        self,
+        inputs: OneOrMany[TDataModel],
+        field_to_embed: str,
+        field_to_store: str,
+        execution_settings: dict[str, "PromptExecutionSettings"],
+        container_mode: bool = False,
+        cast_function: Callable[[list[float]], Any] | None = None,
+        **kwargs: Any,
+    ):
+        """Gather all fields to embed, batch the embedding generation and store."""
+        contents: list[Any] = []
+        dict_like = (getter := getattr(inputs, "get", False)) and callable(getter)
+        list_of_dicts: bool = False
+        if container_mode:
+            contents = inputs[field_to_embed].tolist()  # type: ignore
+        elif isinstance(inputs, list):
+            list_of_dicts = (getter := getattr(inputs[0], "get", False)) and callable(getter)
+            for record in inputs:
+                if list_of_dicts:
+                    contents.append(record.get(field_to_embed))  # type: ignore
+                else:
+                    contents.append(getattr(record, field_to_embed))
+        else:
+            if dict_like:
+                contents.append(inputs.get(field_to_embed))  # type: ignore
+            else:
+                contents.append(getattr(inputs, field_to_embed))
+        vectors = None
+        service: EmbeddingGeneratorBase | None = None
+        for service_id, settings in execution_settings.items():
+            service = self.get_service(service_id, type=EmbeddingGeneratorBase)  # type: ignore
```

Comment:
For my understanding: what mypy error is shown here and why the need for ignore?